src/types/block.rs

```diff
@@ -19,6 +19,7 @@ impl From<BlockHeight> for u32 {
     }
 }
 
+// TODO(evg): avoid code duplicate
```

Comment:
To avoid the code duplication, you could introduce a type parameter for the `Block` struct in the place of `TransactionId` and use the `Block` struct in both return values, just with different type arguments:

- `get_block() -> Result<Result<Block<TransactionId>, RpcError>, HTTPError>`
- `get_block() -> Result<Result<Block<DecodedRawTransaction>, RpcError>, HTTPError>`

Also, while you are at it: Please make all the fields in the `Block` struct public. That is an on-going effort we are trying to achieve (see #17).

Please note that I used `DecodedRawTransaction` here as a result of investigating the error message you got. Apparently, the bitcoin node doesn't return a JSON field `blockhash` when queried for a `verbose` block. The documentation of bitcoin is just bad in this regard :(